### PR TITLE
Use a list of vectors rather than a NaN sepearted list to denote shapes

### LINKED ISSUE #145: Regression: Can no longer set individual node colours
I'll use the example from `README.md` to illustrate the issue:
```jl
using GraphRecipes
using Plots

g = [0 1 1;
     1 0 1;
     1 1 0]

graphplot(g,
          x=[0,-1/tan(π/3),1/tan(π/3)], y=[1,0,0],
          nodeshape=:circle, nodesize=1.1,
          axis_buffer=0.6,
          curves=false,
          color=:black,
          nodecolor=[colorant"#389826",colorant"#CB3C33",colorant"#9558B2"],
          linewidth=10)
```
This no longer produces the expected red, green and purple nodes. Instead, it plots *three green nodes* and prints the following warnings:
```
┌ Warning: Indices Base.OneTo(3) of attribute `fillcolor` does not match data indices 1:242.
└ @ Plots ~/.julia/packages/Plots/SVksJ/src/utils.jl:102
┌ Info: Data contains NaNs or missing values, and indices of `fillcolor` vector do not match data indices.
│ If you intend elements of `fillcolor` to apply to individual NaN-separated segements in the data,
│ pass each segment in a separate vector instead, and use a row vector for `fillcolor`. Legend entries 
│ may be suppressed by passing an empty label.
│ For example,
└     plot([1:2,1:3], [[4,5],[3,4,5]], label=["y" ""], fillcolor=[1 2])
┌ Warning: Indices Base.OneTo(3) of attribute `markercolor` does not match data indices 1:242.
└ @ Plots ~/.julia/packages/Plots/SVksJ/src/utils.jl:102
┌ Info: Data contains NaNs or missing values, and indices of `markercolor` vector do not match data indices.
│ If you intend elements of `markercolor` to apply to individual NaN-separated segements in the data,
│ pass each segment in a separate vector instead, and use a row vector for `markercolor`. Legend entries 
│ may be suppressed by passing an empty label.
│ For example,
└     plot([1:2,1:3], [[4,5],[3,4,5]], label=["y" ""], markercolor=[1 2])
```
Tested with Julia 1.6 and Plots.jl backends GR and PyPlot.

### PR DESCRIPTION
**Pull request by JackDevine**
Fixes #145 also includes some misc tidying changes

### CODE REVIEW

**Comment by BeastyBlacksmith** [comment]
Running the tests using this branch locally on julia 1.0.5 is actually fine on my machine

File: Project.toml [hunk 1]
```diff
@@ -22,7 +22,7 @@ GeometryTypes = "0.7.7, 0.8"
 Interpolations = "0.12.8"
 LightGraphs = "1.3.0"
 NaNMath = "0.3.3"
-NetworkLayout = "0.2.0"
+NetworkLayout = "0.3.0"
```
Comment (BeastyBlacksmith) [inline Project.toml:25]:
```suggestion
NetworkLayout = "0.2.0, 0.3.0"
```

**Comment by BeastyBlacksmith** [comment]
can you rebase on master?

**Comment by BeastyBlacksmith** [comment]
So now we got one version that works on 1.0 and one version that works on 1.6. I'd suggest to set julia compat to 1.6 here.
Unless you want to fix 1.0, which is also fine with me, but I didn't look into that, might require a patch release for Plots 1.4

**Comment by BeastyBlacksmith** [comment]
@JackDevine what do you think? I'd like to get this merged for https://github.com/JuliaPlots/Plots.jl/pull/3519

**Comment by JackDevine** [comment]
Sorry for the long delay.

It is a little unfortunate to drop support for v1.0, but I guess that I am OK with that. Am I right that we are also planning on dropping Julia v1.0 support for StatsPlots too?

**Comment by BeastyBlacksmith** [comment]
As far as I can see StatsPlots never had julia 1.0 in its compat entry and is currently only tested on julia 1.6

**Pull request merged**